### ISSUE TITLE
FunctionUse.ArgumentFunctionsReportCurrentValue.NeedsInspection rule is too strict

### ISSUE THREAD

**Issue by mvorisek** [squizlabs/PHP_CodeSniffer#2528]
`PHPCompatibility.FunctionUse.ArgumentFunctionsReportCurrentValue.NeedsInspection` warning should not be triggered for code like:
```
public function get($v)
{
    if ($v === null) {
        return false;
    }

    $args = func_get_args();

    return $args;
}
```

and if possible also not for PHP functions (for which is known that the parameters are passed by value):
```
public function get($v)
{
    if (is_string($v)) {
        return false;
    }

    $args = func_get_args();

    return $args;
}
```

Similar usage is very common and showing totally unneeded warnings is very undesirable.

**Comment by jrfnl** [comment]
That sniff is not part of PHPCS, but of PHPCompatibility. Please report the issue there.
Or better yet: join the discussion in the existing issue for this: https://github.com/PHPCompatibility/PHPCompatibility/issues/796

**Comment by mvorisek** [comment]
Closing as relevant for another package.
